cmd/web.go

```diff
@@ -69,6 +71,34 @@ func runWeb(ctx *cli.Context) error {
 	if ctx.IsSet("port") {
 		setting.AppURL = strings.Replace(setting.AppURL, setting.HTTPPort, ctx.String("port"), 1)
 		setting.HTTPPort = ctx.String("port")
+
+		switch setting.Protocol {
+		case setting.UnixSocket:
```

Comment:
This will not work for unix sockets and FCGI (see `NewContext` in `setting.go`). 
As being so close to that functionality as in NewContext, can this formatting be moved out to some common place so that it could be reused in both places?

Reply:
@lafriks, for unix sockets and FCGI, the port changed will not be effected, so ignored them. Currently only this place need the code and there is no other place need it.

Reply:
Sorry it was late and forgot about golang cases not being fall through by default :)